homeassistant/components/generic_water_heater/water_heater.py

```diff
@@ -0,0 +1,228 @@
+"""Support for generic water heater units."""
+import logging
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterEntity,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_TEMPERATURE,
+    CONF_NAME,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import DOMAIN as HA_DOMAIN, callback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from . import CONF_HEATER, CONF_SENSOR, CONF_TARGET_TEMP, CONF_TEMP_DELTA
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS_HEATER = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+DEFAULT_NAME = "Generic Water Heater"
+
+
+async def async_setup_platform(
+    hass, hass_config, async_add_entities, discovery_info=None
+):
+    """Set up the generic water_heater devices."""
+    entities = []
+
+    for config in discovery_info:
+        name = config.get(CONF_NAME)
+        heater_entity_id = config.get(CONF_HEATER)
+        sensor_entity_id = config.get(CONF_SENSOR)
```

Comment:
```suggestion
        name = config[CONF_NAME]
        heater_entity_id = config[CONF_HEATER]
        sensor_entity_id = config[CONF_SENSOR]
```

I think these are required?